lib/Pass/Pass.cpp

```diff
@@ -374,6 +374,7 @@ struct SouperPass : public ModulePass {
     }
 
     for (auto &B : CS.Blocks) {
+      CandidateMap CandMap;
```

Comment:
can you explain why this map had to move inside this loop? does it have any other effects that we should think about?

Reply:
The original program loops the CS.Blocks, push all the replacement candidates in each block into CandMap. After all the candidates are pushed into CandMap, then do the replacements. By doing this these candidates loses the information of their original block, and later in replacement stage it cannot tell which use it should be replaced. 
This patch does the replacement in the block loop. CandMap gets refreshed for each Block, and the corresponding uses can now be replaced correctly.
I don't think there is any other side effect.

Reply:
Actually some same replacements might be listed more than once even it is harvested from defs. I'm fixing this.